File: backend/src/main/java/de/nordbyte/mavazihub/returnrequest/entity/ReturnRequest.java
package de.nordbyte.mavazihub.returnrequest.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "return_request")
@Getter
@Setter
public class ReturnRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(name = "order_id", nullable = false)
    private UUID orderId;

    @Column(name = "customer_id", nullable = false)
    private UUID customerId;

    @Column(length = 500)
    private String reason;

    /**
     * Status: REQUESTED | APPROVED | REJECTED | COMPLETED
     * Initialer Status laut Pflichtenheft (UC-RS-01): "beantragt" (REQUESTED)
     */
    @Column(nullable = false, length = 50)
    private String status;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "returnRequest", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<ReturnItem> items = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        if (this.status == null) {
            this.status = "REQUESTED";
        }
    }
}

File: frontend/frontend/src/types/returnRequest.ts
import type { ISODateTimeString, UUID } from './common'

export type ReturnItemRequest = {
  orderItemId: UUID
  quantity: number
}

export type ReturnRequestCreateDTO = {
  orderId: UUID
  reason?: string
  items: ReturnItemRequest[]
}

export type ReturnItemResponse = {
  orderItemId: UUID
  quantity: number
}

export type ReturnRequest = {
  id: UUID
  orderId: UUID
  reason: string
  status: string
  createdAt: ISODateTimeString
  items: ReturnItemResponse[]
}

export type ReturnRequestResponseDTO = ReturnRequest
